nx-X11/programs/Xserver/hw/nxagent/Reconnect.c

```diff
@@ -438,22 +438,10 @@ Bool nxagentReconnectSession(void)
 
   if (nxagentKeyboard != NULL)
   {
-    int size;
+    nxagentOldKeyboard = strndup(nxagentKeyboard, strlen(nxagentKeyboard));
+    /* FIXME: check if nxagentOldKeyboard == NULL and abort */
```

Comment:
No, what this is about is strndup(). It performs an malloc. If that fails strndup will return NULL and _this_ should be checked.